app.py

```diff
@@ -582,15 +580,26 @@ def update_table(total_games, start_date, end_date):
         (start_date, end_date),
     )
     columns = [i[0] for i in cursor.description]
+    replay_server_base_url = "https://replays.wesnoth.org"
     df = (
         pd.DataFrame(cursor.fetchall(), columns=columns)
         .map(lambda x: x[0] if type(x) is bytes else x)
         .assign(
             START_TIME=lambda x: pd.to_datetime(x["START_TIME"]),
             END_TIME=lambda x: pd.to_datetime(x["END_TIME"]),
-            # Calculate the game duration in minutes.
             GAME_DURATION=lambda x: (x["END_TIME"] - x["START_TIME"]).dt.total_seconds()
-            / 60,
+            / 60,  # Calculate the game duration in minutes.
+            REPLAY_NAME=lambda x: "["
+            + x["REPLAY_NAME"]
+            + "]("
+            + replay_server_base_url
+            + "/"
+            + x["INSTANCE_VERSION"].str.slice(0, 4)
+            + "/"
+            + x["END_TIME"].dt.strftime("%Y/%m/%d")
+            + "/"
+            + x["REPLAY_NAME"]
+            + ")",  # Markdown link to replay file download
```

Comment:
While the replay filename is not entirely arbitrary this essentially concatenates arbitrary data to hopefully get the desired markdown syntax. In general this kind of injection should be avoided but probably can't here.

If there is a convenient way to properly generate a markdown link that would be preferable. Otherwise need to check what can happen in the worst case. Given that brackets are currently not allowed in a replay filename probably not too much.

Of course another option would be to just not generate markdown and leave the url as a plain string.

(Adding this as proper review comment, https://github.com/wesnoth/wesnoth-multiplayer-data-dashboard/commit/3d6b0e6834d12927be85a310a29c3ce5b450c95d#comments was easy to miss.)